File: python/check_if_n_and_its_double_exist_1346.py
class Solution:
    def checkIfExist(self, arr: list[int]) -> bool:
        seen: set[int] = set()
        for num in arr:
            if num * 2 in seen or (num % 2 == 0 and num // 2 in seen):
                return True
            seen.add(num)
        return False

File: rust/src/check_if_n_and_its_double_exist_1346.rs
use std::collections::HashSet;

use crate::Solution;

impl Solution {
    pub fn check_if_exist(arr: Vec<i32>) -> bool {
        let mut seen: HashSet<i32> = HashSet::new();
        for num in arr {
            if seen.contains(&(num * 2)) || (num % 2 == 0 && seen.contains(&(num / 2))) {
                return true;
            }
            seen.insert(num);
        }
        false
    }
}
